src/javascript/app_2/App/Components/Elements/AccountSwitcher/__tests__/upgrade_button.spec.js

```diff
@@ -0,0 +1,18 @@
+import React                  from 'react';
+import { expect }             from 'chai';
+import { configure, shallow } from 'enzyme';
+import Adapter                from 'enzyme-adapter-react-16';
+import { UpgradeButton }      from '../upgrade_button.jsx';
+
+configure({ adapter: new Adapter() });
+
+describe('UpgradeButton', () => {
+    it('should render one <UpgradeButton /> component', () => {
+        const wrapper = shallow(<UpgradeButton />);
+        expect(wrapper).to.have.length(1);
+    });
+    it('should have onClick as an instance of function when passed', () => {
+        const wrapper = shallow(<UpgradeButton onClick={() => true} />);
+        expect(wrapper.find('Button').prop('onClick')).to.be.an.instanceof(Function);
```

Comment:
When we are testing against the prop-types, we need to make sure that proper errors are thrown upon getting error. Since you are passing onClick as a function, and then testing the same behavior, we can't really make sure if underlying prop validations are in place or not. So, what happens when we pass an object, for example, instead of a function? 

I think it is better to test out the PropTypes of the component, and either trust that propTypes are in place and we are good to go, or test the component and stub the console to see the error.